lib/CloudinaryPlugin.js

```diff
@@ -0,0 +1,336 @@
+   * Action: replace_tags
+   *  
+   * Replace all the tags of the given assets by the given tag
+   * The request must contain a tag and at least a public_id
+   * 
+   * If one of both is missing, a bad request error is thrown.
+   * 
+   * You can give several public_ids by setting public_id to an array of string 
+   * Be careful of white space as Cloudinary allows space in the public_ids
+   * 
+   * @param {Request} request 
+   * 
+   * @see https://cloudinary.com/documentation/image_upload_api_reference#tags_method
+   */
+  async replace_tag(request) {
+    const tag = this._getArg(request.input.body, 'tag'),
+      to = this._getArg(request.input.body, 'public_id', true),
+      res = await Cloudinary.uploader.replace_tag(tag, to);
+    this._handleError(to, res.public_ids);
+    return res;
+  }
+
+  /**
+   * Controller: Tags
+   * Action: remove_tag
+   *  
+   * Remove the given tag to the given assets 
+   * The request must contain a tag and at least a public_id
+   * 
+   * If one of both is missing, a bad request error is thrown.
+   *
+   * You can give several public_ids by setting public_id to an array of string 
+   * Be careful of white space as Cloudinary allows space in the public_ids
+   * 
+   * @param {Request} request 
+   * 
+   * @see https://cloudinary.com/documentation/image_upload_api_reference#tags_method
+   */
+  async remove_tag(request) {
+    const tag = this._getArg(request.input.body, 'tag'),
+      to = this._getArg(request.input.body, 'public_id', true),
+      res = await Cloudinary.uploader.remove_tag(tag, to);
+    this._handleError(to, res.public_ids);
+    return res;
+  }
+
+  /**
+   * Controller: Tags
+   * Action: remove_all_tags
+   *  
+   * Remove all tags from the given assets
+   * The request must contain at least a public_id
+   * 
+   * If none is given, a bad request error is thrown.
+   * 
+   * You can give several public_ids by setting public_id to an array of string 
+   * Be careful of white space as Cloudinary allows space in the public_ids
+   * 
+   * @param {Request} request 
+   * 
+   * @see https://cloudinary.com/documentation/image_upload_api_reference#tags_method
+   */
+  async remove_all_tags(request) {
+    const to = this._getArg(request.input.body, 'public_id', true),
+      res = await Cloudinary.uploader.remove_all_tags(to);
+    this._handleError(to, res.public_ids);
+    return res;
+  }
+
+  /** Extracts the args with the given name from the given request. 
+   * If the request does not contains an attribute with the given name, it will throw a bad request error. 
+   * 
+   * If array attribute is equal to true, the function returns an array containing the needed args
+   * 
+   * @param {Request} request 
+   * @param {String} name 
+   */
+  _getArg(source, name, array = false) {
+    if (source[name]) {
+      if (array) {
+        return source[name] instanceof Array ? source[name] : [source[name]];
+      }
+      return source[name];
+    }
```

Comment:
I think it's could be more clear, something like this:
```javascript
if (source[name]) {
  return array && !(source[name] instanceof Array)
    ? [source[name]]
    : source[name]:
}
```

Reply:
You don't like it ?

Reply:
This [has been changed](https://github.com/kuzzleio/kuzzle-plugin-cloudinary/blob/e670167760e7883bb707a0dd4ca30bb0735c392b/lib/CloudinaryPlugin.js#L356)  the moment you gave the advice :) And thank you for that. (Though, i don't understand why the change does not appear here)